packages/pnpm-sync-lib/etc/pnpm-sync-lib.api.md

```diff
@@ -39,6 +51,11 @@ export interface IPnpmSyncCopyOptions {
     // (undocumented)
     getPackageIncludedFiles: (packagePath: string) => Promise<string[]>;
     // (undocumented)
+    logMessageCallback: (message: string, messageKind: ILogMessageKind, details?: {
+        fileCount: number;

```

Comment:
Because `details` is an input to the caller, it would be better to make it always defined and simply pass `{}` if it is empty.  This avoids the caller having to annoyingly test whether it is `undefined`.  (If you don't want to use that parameter at all, the TypeScript type system will not require the caller to provide it.)

On the other hand, rather than defining different interfaces for each API, it might be more flexible to define a single `logMessageCallback` signature that receives a `messageId` to identify the messages.  In this way you could add new message types later, which can have different `details`. The fancy way to do this would be like:
```ts
type MessageDetails = {
  messageId: 'starting-sync'; // discriminator
  sourceFolder: number;
  targetFolder: number;
} | {
  messageId: 'sync-timing';
  totalMs: number;
}

switch (details.messageId) {
  case 'starting-sync':
    console.log(details.sourceFolder);
    break;
}
``` 

but you could also be lazy and just do this:

```ts
interface IMessageDetails {
  sourceFolder?: number;
  targetFolder?: number;
  totalMs?: number;
}

// And then the caller just "knows" which ones should be defined based on your docs
switch (details.messageId) {
  case 'starting-sync':
    console.log(details.sourceFolder!);
    
    // or if you are paranoid
    console.log(details.sourceFolder ?? '???');
    break;
}
```